flocker/control/_protocol.py

```diff
@@ -342,8 +365,14 @@ def node_changed(self, source, state_changes):
         :param list state_changes: One or more ``IClusterStateChange``
             providers representing the state change which has taken place.
         """
-        self.cluster_state.apply_changes_from_source(source, state_changes)
-        self._send_state_to_connections(self.connections)
+        # XXX The IClusterStateSource source doesn't seem terribly useful.  I
+        # hoped it might have an IP address or node UUID which we could log but
+        # it only has `last_activity`. On the other hand there will be a parent
+        # Action that includes information about which connection called this
+        # method remotely.
```

Comment:
Or ... yes, there is.  It's just from another node.  If you examine the merged logs then the log structure does tell you who triggered this.